dev/Controller/SPController.php

```diff
@@ -130,6 +142,31 @@ public function assertionConsumerServiceAction(Request $request)
         }
     }
 
+    private function getServiceProvider(Request $request)
+    {
+        // The MockAzureMfaController test sends the mock sso response directly to the demo sp. Normally the Azure MFA
+        // GSSP SP would be targeted first. In order to prevent SAML errors, the correct SP entity is loaded for this
+        // test.
+        $mockMfaReferer = 'https://azure-mfa.stepup.example.com/mock/sso';
+        if (strstr($request->server->get('HTTP_REFERER'), $mockMfaReferer) !== false) {
```

Comment:
I do not think basing this logic on the environment would be sufficiently specific. The referer is also important as both would probably run in test env.

Reply:
I've removed the addiotnal comlexity. Could you review this and maybe we could discuss IRL if this could be merged back.
https://github.com/OpenConext/Stepup-Azure-MFA/commit/54a854e790287c265c8905f979a16b5648921c0d